src/ucs/config/parser.h

```diff
@@ -65,6 +65,23 @@ typedef struct ucs_config_field {
 } ucs_config_field_t;
 
 
+typedef struct {
+    const char               *name;   /* Key name */
+    const char               *doc;    /* Documentation */
+    size_t                   offset;  /* Storage offset */
```

Comment:
can we use a static array instead of offset? i.e assume all values are stored sequentially, according to the order of keys in the keys array

Reply:
Please correct me if I misunderstood your idea. I didn't understand the role of 'static array' in your proposal.
Currently we have this design:
```
typedef struct {
    int             temp_front;
    int             temp_rear;
} car_opts_t;

  {"TEMP", "20", "Temperature", 0,
    UCS_CONFIG_TYPE_KEY_VALUE(UCS_CONFIG_TYPE_UINT,
        {"front", "front temperature", ucs_offsetof(car_opts_t, temp_front)},
        {"rear",  "rear temperature",  ucs_offsetof(car_opts_t, temp_rear)},
        {NULL}
    )},

```
Now what you propose would be something like:
```
typedef struct {
    int             temp_front;
    int             temp_rear;
// OR you think of this?
    int    temps[2];
} car_opts_t;

  {"TEMP", "20", "Temperature", ucs_offsetof(car_opts_t, temp_front), << offset to the first field
    UCS_CONFIG_TYPE_KEY_VALUE(UCS_CONFIG_TYPE_UINT,
        sizeof(int),  << we also need to store the size of the entry, right?
        {"front", "front temperature"},  << implicit offset 0
        {"rear",  "rear temperature"},  << implicit offset 0 + sizeof(entry)
        {NULL}
    )},

```
Did I get right your idea?

If I get it right, I have some concerns:
1. If we require that storage is always an array (e.g. `int    temps[2];`) then in C code it might not be obvious what temperature we are using by writing `temp[0]`. We might need to come up with enum in this case:
```
typedef enum {
  UCS_CONFIG_TEMP_FRONT,
  UCS_CONFIG_TEMP_REAR,
  CONFIG_TEMP_SIZE
}
int    temps[CONFIG_TEMP_SIZE];
int access = temps[UCS_CONFIG_TEMP_FRONT];
```
2. If we rely on sequential order of fields in the structure - that should work well for primitives, but is it safe regarding alignment or more complex types? There is another issue with this approach: variable access becomes implicit. There is no explicit reference to variable temp_rear. This is error-prone IMO

Also it's not very clear what are benefits of this approach (If I get it right of course)?